src/main/scala/org/dia/algorithms/mcs/MCCOps.scala

```diff
@@ -0,0 +1,237 @@
+      potentialNodeSet: mutable.HashSet[String],
+      visitedNodeSet: mutable.HashSet[String]): mutable.MutableList[mutable.HashSet[String]] = {
+
+    if (visitedNodeSet.contains(node)) {
+      if (globalMCCs.contains(node)) {
+        return globalMCCs(node)
+      }
+      else return new mutable.MutableList[mutable.HashSet[String]]()
+    }
+
+    val isPotentialMCCNode = potentialNodeSet.contains(node)
+    visitedNodeSet.add(node)
+
+    if (outgoingEdgeMap.contains(node)) {
+      for (childNode <- outgoingEdgeMap(node)) {
+        val MCCNodeSet = traversePath(childNode, outgoingEdgeMap, globalMCCs,
+          potentialNodeSet, visitedNodeSet)
+        /** If current node is a potential node then add it to the paths returned bu its children */
+        if (isPotentialMCCNode) {
+          for (path <- MCCNodeSet) {
+            if (path.size > 0) {
+              val tempSet = new mutable.HashSet[String]()
+              tempSet ++= path
+              tempSet += node
+              if (globalMCCs.contains(node)) {
+                globalMCCs(node) += tempSet
+              }
+              else {
+                val tempList = new mutable.MutableList[mutable.HashSet[String]]()
+                tempList += tempSet
+                globalMCCs(node) = tempList
+              }
+            }
+          }
+        }
+      }
+    }
+    if (isPotentialMCCNode) {
+      if (!globalMCCs.contains(node)) {
+        val path = new mutable.HashSet[String]()
+        path += node
+        val tempList = new mutable.MutableList[mutable.HashSet[String]]()
+        tempList += path
+        globalMCCs(node) = tempList
+      }
+      return globalMCCs(node)
+    }
+    else return new mutable.MutableList[mutable.HashSet[String]]()
+  }
+
+  /**
+   * Check if a given node satisfies the two criterions specified by Laurent et al 1998
+   * @param node The MCS Node to check
+   * @param areaCriteriaA
+   * @param areaCriteriaB
+   * @param tempCriteriaA
+   * @param tempCriteriaB
+   * @return A tuple with a boolean indicating if the node satisfies the criteria or not
+   */
+  def isPotentialNode(node : MCSNode,
+      areaCriteriaA: Int,
+      areaCriteriaB: Int,
+      tempCriteriaA: Float,
+      tempCriteriaB: Float):
+      (Boolean, String) = {
+
+    /** Area within the node matching the temp criteria for A acc to Laurent et al 1998 */
+    var nodeAreaCriteriaA: Int = 0
+    /** Area within the node matching the temp criteria for B acc Laurent et al 1998 */
+    var nodeAreaCriteriaB: Int = 0
+
+    for ((pos, temp) <- node.grid) {
+      // Acc to Laurent et al temp for criteria ! is less than criteria B
+      if (temp <= tempCriteriaA) {
+        nodeAreaCriteriaA += 1
+        nodeAreaCriteriaB += 1
+      }
+      else if (temp <= tempCriteriaB) {
+        nodeAreaCriteriaB += 1
+      }
+    }
+    if (nodeAreaCriteriaA >= areaCriteriaA || nodeAreaCriteriaB >= areaCriteriaB) {
```

Comment:
This if statement should be 
if (nodeAreaCriteriaA >= areaCriteriaA && nodeAreaCriteriaB >= areaCriteriaB)